backend/internal/service/matching/engine.go

```diff
@@ -8,20 +8,47 @@ import (
 	"day-trading-app/backend/internal/service/models"
 	"github.com/google/uuid"
 	"strings"
+	"sync"
 	"time"
 
+	"github.com/adrianbrad/queue"
 	"github.com/ryszard/goskiplist/skiplist"
 )
 
+type StockMatch struct {
+	Order       models.StockTransaction `json:"order" bson:"order"`                 // original order; though matching engine will change OrderStatus
+	QuantityTx  int                     `json:"quantity_tx" bson:"quantity_tx"`     // quantity actually transacted
+	PriceTx     int                     `json:"price_tx" bson:"price_tx"`           // price actually transacted
+	CostTotalTx int                     `json:"cost_total_tx" bson:"cost_total_tx"` // total cost transacted; needed for parent tx
+	IsParent    bool                    `json:"is_parent" bson:"is_parent"`         // true if transaction has created a child
+	Killed      bool                    `json:"killed" bson:"killed"`               // expired or cancelled
+}
+
+func init() {
+	// Start ticker goroutine to periodically flush expired limit orders
+	go func() {
+		for range time.Tick(1 * time.Second) {
+			FlushExpired()
+		}
+	}()
+}
```

Comment:
this looks very clean 👏

Reply:
future us may need to evaluate what would happen if FlushExpired is called before its last execution finishes. may cause race conditions with the Peek and Get functions on the queue